src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -221,14 +221,17 @@ private function handleException(\Exception $e, Request $request, int $type): Re
         // a listener might have replaced the exception
         $e = $event->getException();
 
-        if (!$event->hasResponse()) {
+        if ($event->hasResponse()) {
+            $response = $event->getResponse();
+        } elseif ($e instanceof HttpExceptionInterface) {
+            $code = $e->getStatusCode();
+            $response = new Response(isset(Response::$statusTexts[$code]) ? $code.' '.Response::$statusTexts[$code] : $code);
```

Comment:
Would it make sense to do this only in non-debug mode? And keep the exception in debug mode?

Reply:
:+1: for keeping them in debug

Reply:
Nevermind my prior comment. You already answered it in https://github.com/symfony/symfony/pull/26928#issuecomment-383074122. Please ignore. :-)